src/main/java/org/opentripplanner/datastore/https/HttpsDataSourceRepository.java

```diff
@@ -0,0 +1,60 @@
+package org.opentripplanner.datastore.https;
+
+import java.net.URI;
+import javax.annotation.Nonnull;
+import org.opentripplanner.datastore.api.CompositeDataSource;
+import org.opentripplanner.datastore.api.DataSource;
+import org.opentripplanner.datastore.api.FileType;
+import org.opentripplanner.datastore.base.DataSourceRepository;
+import org.opentripplanner.datastore.base.ZipStreamDataSourceDecorator;
+
+/**
+ * This data store accesses files in read-only mode over HTTPS.
+ */
+public class HttpsDataSourceRepository implements DataSourceRepository {
+
+  @Override
+  public String description() {
+    return "HTTPS";
+  }
+
+  @Override
+  public void open() {}
+
+  @Override
+  public DataSource findSource(@Nonnull URI uri, @Nonnull FileType type) {
+    if (skipUri(uri)) {
+      return null;
+    }
+    return createSource(uri, type);
+  }
+
+  @Override
+  public CompositeDataSource findCompositeSource(@Nonnull URI uri, @Nonnull FileType type) {
+    if (skipUri(uri)) {
+      return null;
+    }
+    return createCompositeSource(uri, type);
+  }
+
+  /* private methods */
+
+  private static boolean skipUri(URI uri) {
+    return !"https".equals(uri.getScheme());
```

Comment:
I doesn't happen a lot but do we not want to support HTTP without TLS?

Reply:
I would rather not implement it right away (for security reason). This could be easily added afterwards if someone asks for it.